multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ApplicationWaitAfterStopVariableGetter.java

```diff
@@ -0,0 +1,59 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.time.Duration;
+
+import org.cloudfoundry.multiapps.controller.core.util.ApplicationConfiguration;
+import org.cloudfoundry.multiapps.controller.process.Constants;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.steps.ProcessContext;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.springframework.util.ObjectUtils;
+
+import com.sap.cloudfoundry.client.facade.domain.CloudApplication;
+
+public class ApplicationWaitAfterStopVariableGetter {
+
+    public static boolean isStopDelayVariableSet(ProcessContext context) {
+        String stopDelayFromEnv = extractDelayVariableFromEnv(context);
+        return !ObjectUtils.isEmpty(stopDelayFromEnv);
+    }
+
+    private static String extractDelayVariableFromEnv(ProcessContext context) {
+        CloudApplication app = context.getVariable(Variables.APP_TO_PROCESS);
+        return app.getEnv()
+                  .get(Constants.VAR_WAIT_AFTER_APP_STOP);
+    }
+
+    public static Duration getDelayDurationFromAppEnv(ProcessContext context) {
+        String stopDelayFromEnv = extractDelayVariableFromEnv(context);
+        if (ObjectUtils.isEmpty(stopDelayFromEnv)) {
+            return Duration.ZERO;
+        }
+        return processDurationFromAppEnv(stopDelayFromEnv, context);
+    }
+
+    private static Duration processDurationFromAppEnv(String stopDelayFromEnv, ProcessContext context) {
+        try {
+            int delay = Integer.parseInt(stopDelayFromEnv);
+            return convertDelayToSeconds(delay, context);
+        } catch (NumberFormatException e) {
+            context.getStepLogger()
+                   .debug(Messages.ERROR_WHILE_READING_APP_STOP_DELAY, e.getMessage());
```

Comment:
Warn or Error?

Reply:
I think that changing the debug to warn might be more appropriate. In addition I'll rename the message constant name from "ERROR_..." to something else to avoid confusion.